Add SPDX-License-Identifier to source files

prasannakotyal opened the pull request:
## Why are these changes needed?
Add SPDX-License-Identifier to source files

- Adds SPDX-License-Identifier: Apache-2.0 to all source files
- Places identifier at the beginning of files (after shebang if present)
- Skips files that already have SPDX identifiers
- Handles different file types and comment styles
## Related issue number (if any).
Fixes #1311

prasannakotyal committed:
Add SPDX-License-Identifier to source files

Signed-off-by: prasannakotyal <prasannakotyal@gmail.com>

shahrokhDaijavad commented:
Hi, @prasannakotyal. As you can see, after merging the latest dev branch into your branch, all CI/CD checks have passed! One request that I have is to modify your script so that the SPX-License-identifier line is NOT added to the yaml files. Many of our yaml files get created dynamically, and we are not required to add the identifier to these files.
The current repo (as of now) has 746 .py files, and those are the only ones that need the identifier line.

prasannakotyal commented:
HI, @shahrokhDaijavad. I understand, I'll modify the script to remove the adding of  SPX-License-identifier to the YAML files.
Thanks for the update.

prasannakotyal committed:
Update script to only process Python files and remove YAML handling

Signed-off-by: prasannakotyal <prasannakotyal@gmail.com>

prasannakotyal commented:
Hi @shahrokhDaijavad, I have made the necessary changes now:
Update SPDX identifier script to only process Python files

Changes:
- Modified script to only add SPDX identifiers to Python (.py) files
- Removed support for YAML and other file types
- Maintains existing functionality for Python files (adding after shebang if present)

Thanks

shahrokhDaijavad commented:
@prasannakotyal Looking at the latest changes you have in the branch, it looks as if all .yaml and .sh files still have the "SPDX-License-Identifier: Apache-2.0" line at the top. For example, look at all the yaml files under `.github/workflows` folder or all shell scripts under `scripts `folder

prasannakotyal committed:
Remove SPDX license identifier from non-Python files

This commit removes the SPDX license identifier line from all non-Python files (e.g., .yaml, .yml, and .sh) to ensure consistency across the codebase.

Signed-off-by: prasannakotyal <prasannakotyal@gmail.com>

prasannakotyal committed:
This commit removes the SPDX license identifier line from all non-Python files (e.g., .yaml, .yml, and .sh) to ensure consistency across the codebase."

Signed-off-by: prasannakotyal <prasannakotyal@gmail.com>

prasannakotyal commented:
Hi, @shahrokhDaijavad. I had a small misunderstanding there. I have fixed it now. I have removed the "SPDX-License-Identifier: Apache-2.0" line from all non .py files and should now be proper.

shahrokhDaijavad commented:
Thanks, @prasannakotyal. It looks good to me. 747 .py files and 748 occurrences of the "SPDX-License-Identifier: Apache-2.0", because it appears twice in your script file. That looks right to me. 

@touma-I I am approving the PR. Please run the privileged tests and review. Thanks.

shahrokhDaijavad reviewed (approved):
LGTM.

swith005 reviewed (approved):
lgtm - thanks!

shahrokhDaijavad commented:
@prasannakotyal As you can see, our reviewers have approved your PR, however, if you update the branch, it will run into conflicts with the .py files that have changed since you submitted the PR. I may be wrong, but the easiest solution may be to update your branch to the latest dev and run your script again. Sorry for the extra work, but when a PR affects so many files, this is inevitable.

prasannakotyal committed:
Merge branch 'dev' into add-spdx-identifier

shahrokhDaijavad commented:
Wow! Looks like no conflicts. 
@touma-I  Looks like the only failures are the privileged workflows. Can we merge after CI/CD, before we start changing .py files?
